File: dizi-cafe-frontend/src/services/filmLikeService.ts
import api from "./api"; // token interceptor'lı axios instance

const BASE_URL = "http://localhost:8080/diziCafe/film-likes";

export const likeFilm = async (filmId: number) =>
  api.post(`${BASE_URL}/like/${filmId}`);

export const dislikeFilm = async (filmId: number) =>
  api.post(`${BASE_URL}/dislike/${filmId}`);

export const removeReaction = async (filmId: number) =>
  api.delete(`${BASE_URL}/remove/${filmId}`);

export const hasLiked = async (filmId: number) =>
  api.get<boolean>(`${BASE_URL}/hasLiked/${filmId}`);

export const hasDisliked = async (filmId: number) =>
  api.get<boolean>(`${BASE_URL}/hasDisliked/${filmId}`);

export const getLikeCount = async (filmId: number) =>
  api.get<number>(`${BASE_URL}/likeCount/${filmId}`);

export const getDislikeCount = async (filmId: number) =>
  api.get<number>(`${BASE_URL}/dislikeCount/${filmId}`);

export const getMyLikedFilms = async () =>
  api.get(`${BASE_URL}/my-likes`);

export const getMyDislikedFilms = async () =>
  api.get(`${BASE_URL}/my-dislikes`);

File: diziCafe-backend/src/main/java/com/diziCafe/service/FilmLikeService.java
package com.diziCafe.service;

import com.diziCafe.model.Film;
import com.diziCafe.model.FilmLike;

import java.util.List;

public interface FilmLikeService {

    void likeFilm(Long filmId);

    void dislikeFilm(Long filmId);

    void removeReaction(Long filmId);

    boolean hasLiked(Long filmId);

    boolean hasDisliked(Long filmId);

    long getLikeCount(Long filmId);

    long getDislikeCount(Long filmId);

    List<Film> getUserLikedFilms();

    List<Film> getUserDislikedFilms();
}
